src/JustSaying.Extensions.DependencyInjection.Microsoft/IServiceCollectionExtensions.cs

```diff
@@ -165,14 +166,16 @@ public static IServiceCollection AddJustSaying(this IServiceCollection services,
                 (serviceProvider) =>
                 {
                     var builder = serviceProvider.GetRequiredService<MessagingBusBuilder>();
-                    return builder.BuildPublisherAsync().GetAwaiter().GetResult();
+                    return Task.Run(() => builder.BuildPublisherAsync())
```

Comment:
What's the reasoning for this change? Isn't `Task.Run()` to do a synchronous call an anti-pattern? I'm sure we've had production incidents from this in the past.

Reply:
I'm currently seeing a deadlock in this test: `WhenRegisteringAHandlerViaContainerWithMissingRegistration.Then_An_Exception_Is_Thrown`. Stu suggested this might fix it but it still appears to be deadlocking when resolving `IEnumerable<IHandlerAsync<T>>` in the `ServiceProviderResolver` when resolving an `IMessagingBus`. 

I'm now seeing if I can easily separate the async queue initialization from queue creation at the moment, so we could add the async tasks to a collection of tasks we execute on bus startup. That way, the DI wireup would be entirely sync which is our goal here.

Reply:
Here's a PR that resolves this issue, and also resolves #452 https://github.com/gkinsman/JustSaying/pull/52